### ISSUE TITLE
New product configuration should default to use "plug-ins and features" 

### ISSUE THREAD

**Issue by vogella** [eclipse-pde/eclipse.pde#761]
A newly created product via the PDE wizard should use "plug-ins and features" as default, as this is the most flexible way of defining products. 

![training_product32](https://github.com/eclipse-pde/eclipse.pde/assets/139910/67f7b2a1-07bf-4a32-9df2-24b4531d48b0)

(Highlight is wrong but I think @HannesWell will still know what I mean).

**Comment by HannesWell** [comment]
> A newly created product via the PDE wizard should use "plug-ins and features" as default, as this is the most flexible way of defining products.

I would be happy with that. Lets target that for M3, just in case major problems are reported for mixed products.

In general I already contemplated to drop support for `plug-in only` and `feature only` products and just make all products mixed. Every user is free to fill them only with Plug-ins or only with Features and I see no reason this choice has to be done in advance.
The same applies for run-configurations (although `Feature-based `configs are already somehow mixed because you can add Plugins as well). This would also simplify the UI and would be a good opportunity to check if the version inclusion rules, could be simplified. Currently those rules are relatively complicated and a bit different for Plug-in and Feature based launches considers if the version listed in the config matches a Plugin in the TP or workspace and if it has a the same major and minor, but the micor-version can be different. It is convoluted.

**Comment by laeubi** [comment]
> In general I already contemplated to drop support for plug-in only and feature only products and just make all products mixed. 

Just keep in mind that then PDE Build most likely can't build these products as only P2 / Tycho supports mixed products, also exporting such products from the UI will likely not work anymore.

But in general a product should neither have _features_  or _plugins_ at all (mixed or not) it should contain _requirements_ what the could be anything (plugins, features, resources, binaries, ....)

**Comment by HannesWell** [comment]
> > In general I already contemplated to drop support for plug-in only and feature only products and just make all products mixed.
> 
> Just keep in mind that then PDE Build most likely can't build these products as only P2 / Tycho supports mixed products, also exporting such products from the UI will likely not work anymore.

Yes that's right, but at least for the PDE export from the workspace I have the intention to replace it by direct calls to the P2-director etc. one day instead of generating ANT scripts that use the PDE-Build tasks.
But personally I have no interest in maintaining the PDE-Build ANT tasks themself.
Unfortunately from other current discussions I see that there still seem to be active usage of them.
@iloveeclipse are you still using the PDE-build ANT tasks?

**Comment by iloveeclipse** [comment]
Is this still targeted for 4.30 M3?

**Comment by HannesWell** [comment]
> Is this still targeted for 4.30 M3?

At least I forgot it.
Lets re-target this for 4.31-M1. I don't think it is urgent.
